portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -17,31 +17,59 @@
 import com.google.gson.Gson;
 import java.util.Date;
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
-public class DataServlet extends HttpServlet {
+// /** Servlet that returns some example content. TODO: modify this file to handle comments data */
+// @WebServlet("/data")
+// public class DataServlet extends HttpServlet {
 
+//     
+// }
+
+
+/** Servlet that processes text. */
+@WebServlet("/text")
+public final class DataServlet extends HttpServlet {
+    //modify DataServlet to contain an ArrayList<String> variable
+    
+    private ArrayList<String> allComments = new ArrayList<String>();
+    
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        
-        //modify DataServlet to contain an ArrayList<String> variable
-        ArrayList<String> messages = new ArrayList<String>();
-        messages.add("This is the first message");
-        messages.add("This is the second message");
-        messages.add("This is the third message");
-        
+        response.setContentType("text/html");
         //convert to Json using Gson
-        Gson gson = new Gson();
-        String json = gson.toJson(messages);
+        String json = new Gson().toJson(allComments);
+        response.getWriter().println(json);
+    }
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String text = getParameter(request, "text-input", "");
 
+        // add to comment collection
+        allComments.add(request.getParameter("text-input"));
+
+        // Respond with the result.
         response.setContentType("text/html;");
-        response.getWriter().println(json);
+        response.getWriter().println(text);
+        response.sendRedirect("/commentPage.html");
```

Comment:
This is honestly something I'm just curious about. What's the advantage of both writing the new comments in the response and redirecting them to the commentPage? I'd sort of expect one or the other. I'd love to learn why both are needed :)

Reply:
I could be wrong, but I'm pretty sure response.sendRedirect("/commentPage.html"); redirects the user to the comment page after they submit their comment. It automatically redirects them so that they don't end up being taken to an automatic un-stylized white page that just has their submitted message.